connectors/sources/google_cloud_storage.py

```diff
@@ -199,10 +201,50 @@ async def _api_call(
                 )
                 await asyncio.sleep(DEFAULT_WAIT_MULTIPLIER**retry_counter)
 
+    def get_pem_format(self, private_key, max_split=-1):
+        """Convert key into PEM format.
+
+        Args:
+            private_key (str): Private_key in raw format.
+            max_split (int): Specifies how many splits to do. Defaults to -1.
+
+        Returns:
+            string: PEM format
+        """
+        private_key = private_key.replace(" ", "\n")
+        private_key = " ".join(private_key.split("\n", max_split))
+        private_key = " ".join(private_key.rsplit("\n", max_split))
+        return private_key
+
+    def _initialize_configurations(self):
+        """Initialize the ServiceAccountCreds"""
+        if self.service_account_credentials is not None:
+            return
+
+        json_credentials = json.loads(self.configuration["service_account_credentials"])
+
+        if (
+            json_credentials.get("private_key")
+            and "\n" not in json_credentials["private_key"]
+        ):
+            json_credentials["private_key"] = self.get_pem_format(
+                private_key=json_credentials["private_key"].strip(),
+                max_split=2,
+            )
+
+        self.service_account_credentials = ServiceAccountCreds(
+            scopes=[CLOUD_STORAGE_READ_ONLY_SCOPE],
+            **json_credentials,
+        )
+        self.user_project_id = self.service_account_credentials.project_id
+
     async def ping(self):
         """Verify the connection with Google Cloud Storage"""
+
         if RUNNING_FTEST:
             return
+
+        self._initialize_configurations()
```

Comment:
Shouldn't we initialize configurations in the constructor and not in `ping`?

Reply:
To initialize `ServiceAccountCreds` we need to validate the Json first which will be done in validate_config. Thus, Updated the `self._initialize_configurations()` as a getter method, with a check to return if the `ServiceAccountCreds` is already initialized.